angular/projects/spark-angular/src/lib/components/sprk-autocomplete/sprk-autocomplete.component.ts

```diff
@@ -0,0 +1,421 @@
+          this.input.ref.nativeElement,
+          'aria-activedescendant',
+          listItemElement.ref.nativeElement.id,
+        );
+      }
+    }
+  }
+
+  /**
+   * Emit the itemSelectedEvent with the currently-highlighted Autocomplete
+   * result item.
+   */
+  selectHighlightedListItem(): void {
+    var selectedElement = this.resultItems.filter(
+      (element) => element.isSelected,
+    )[0];
+    var selectedId = selectedElement.ref.nativeElement.id;
+    this.itemSelectedEvent.emit(selectedId);
+  }
+
+  /**
+   * Hide the Autocomplete results list.
+   */
+  hideResults(): void {
+    if (this.results) {
+      // Remove the hidden style
+      this.renderer.addClass(
+        this.results.nativeElement,
+        'sprk-c-Autocomplete__results--hidden',
+      );
+
+      if (this.input) {
+        // Set aria-expanded on the input's parent
+        this.renderer.setAttribute(
+          this.input.ref.nativeElement.parentNode,
+          'aria-expanded',
+          'false',
+        );
+
+        // Remove aria-activedescendant from the input
+        this.renderer.removeAttribute(
+          this.input.ref.nativeElement,
+          'aria-activedescendant',
+        );
+      }
+
+      // remove aria-selected and highlight class from all list items
+      this.removeAllHighlights();
+
+      this.closedEvent.emit();
+      this.isOpen = false;
+    }
+  }
+
+  /**
+   * Show the Autocomplete results list.
+   */
+  showResults(): void {
+    if (this.results) {
+      this.renderer.removeClass(
+        this.results.nativeElement,
+        'sprk-c-Autocomplete__results--hidden',
+      );
+
+      if (this.input) {
+        // Set aria-expanded on the input's parent
+        this.renderer.setAttribute(
+          this.input.ref.nativeElement.parentNode,
+          'aria-expanded',
+          'true',
+        );
+      }
+
+      this.openedEvent.emit();
+      this.isOpen = true;
+    }
+  }
+
+  /**
+   * @ignore
+   * Track the index of the currently-highlighted Autocomplete result item.
+   */
+  highlightedIndex = -1;
+
+  /**
+   * @ignore
+   */
+  ngAfterContentInit(): void {
+    if (this.results) {
+      if (this.isOpen) {
+        this.showResults();
+      } else {
+        this.hideResults();
+      }
+
+      if (this.input) {
+        // set aria-controls on the input to the id of the results
+        this.generateAriaControls(
+          this.input.ref.nativeElement,
+          this.results.nativeElement,
+        );
+
+        // set aria-owns on the input's parent to the id of the results
+        this.generateAriaOwns(
+          this.input.ref.nativeElement.parentNode,
+          this.results.nativeElement,
+        );
+      }
+    }
+
+    // if itemSelectedEvent is specified, also set that as the click event on each result
+    if (this.itemSelectedEvent) {
+      this.resultItems.forEach((element) => {
+        element.clickedEvent = this.itemSelectedEvent;
+      });
+    }
```

Comment:
This might need to move to another lifecycle hook. It looks like this is not being applied when the component is added/removed dynamically.